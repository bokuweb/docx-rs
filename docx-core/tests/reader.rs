use insta::assert_json_snapshot;

use docx_rs::*;
use std::fs::*;
use std::io::{Read, Write};

#[test]
pub fn read_hello() {
    let mut file = File::open("../fixtures/hello_world/hello_world.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_hello", &json);

    let path = std::path::Path::new("./tests/output/hello.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_numbering() {
    let mut file = File::open("../fixtures/numbering/numbering.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_numbering", &json);

    let path = std::path::Path::new("./tests/output/numbering.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_decoration() {
    let mut file = File::open("../fixtures/decoration/decoration.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_decoration", &json);

    let path = std::path::Path::new("./tests/output/decoration.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_highlight_and_underline() {
    let mut file =
        File::open("../fixtures/highlight_and_underline/highlight_and_underline.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_highlight_and_underline", &json);

    let path = std::path::Path::new("./tests/output/highlight_and_underline.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_history() {
    let mut file = File::open("../fixtures/history_libre_office/history.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_history", &json);

    let path = std::path::Path::new("./tests/output/history.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_indent_word_online() {
    let mut file = File::open("../fixtures/indent_word_online/indent.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_indent_word_online", &json);

    let path = std::path::Path::new("./tests/output/indent_word_online.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_tab_and_break() {
    let mut file = File::open("../fixtures/tab_and_break/tab_and_break.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_tab_and_break", &json);

    let path = std::path::Path::new("./tests/output/tab_and_break.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_table_docx() {
    let mut file = File::open("../fixtures/table_docx/table.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_table_docx", &json);

    let path = std::path::Path::new("./tests/output/table_docx.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_table_merged_libre_office() {
    let mut file = File::open("../fixtures/table_merged_libre_office/table_merged.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_table_merged_libre_office", &json);

    let path = std::path::Path::new("./tests/output/table_merged_libre_office.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_bom() {
    let mut file = File::open("../fixtures/bom/bom.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_bom", &json);

    let path = std::path::Path::new("./tests/output/bom.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_bookmark() {
    let mut file = File::open("../fixtures/bookmark/bookmark.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_bookmark", &json);

    let path = std::path::Path::new("./tests/output/bookmark.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_insert_table() {
    let mut file = File::open("../fixtures/insert_table/insert_table.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_insert_table", &json);

    let path = std::path::Path::new("./tests/output/insert_table.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_textbox() {
    let mut file = File::open("../fixtures/textbox/textbox.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_textbox", &json);

    let path = std::path::Path::new("./tests/output/textbox.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_from_doc() {
    let mut file = File::open("../fixtures/from_doc/from_doc.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_from_doc", &json);

    let path = std::path::Path::new("./tests/output/from_doc.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_lvl_override() {
    let mut file = File::open("../fixtures/lvl_override/override.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_lvl_override", &json);

    let path = std::path::Path::new("./tests/output/lvl_override.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_comment() {
    let mut file = File::open("../fixtures/comment/comment.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_comment", &json);

    let path = std::path::Path::new("./tests/output/comment.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_extended_comment() {
    let mut file = File::open("../fixtures/extended_comments/extended_comments.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("read_extended_comments", &json);

    let path = std::path::Path::new("./tests/output/extended_comments.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}

#[test]
pub fn read_line_spacing() {
    let mut file = File::open("../fixtures/line_spacing/line_spacing.docx").unwrap();
    let mut buf = vec![];
    file.read_to_end(&mut buf).unwrap();
    let json = read_docx(&buf).unwrap().json();

    assert_json_snapshot!("line_spacing", &json);

    let path = std::path::Path::new("./tests/output/line_spacing.json");
    let mut file = std::fs::File::create(&path).unwrap();
    file.write_all(json.as_bytes()).unwrap();
    file.flush().unwrap();
}
