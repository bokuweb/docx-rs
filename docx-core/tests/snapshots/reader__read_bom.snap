---
source: docx-core/tests/reader.rs
expression: "&json"
---
"{\n  \"contentType\": {\n    \"types\": {\n      \"/_rels/.rels\": \"application/vnd.openxmlformats-package.relationships+xml\",\n      \"/docProps/app.xml\": \"application/vnd.openxmlformats-officedocument.extended-properties+xml\",\n      \"/docProps/core.xml\": \"application/vnd.openxmlformats-package.core-properties+xml\",\n      \"/docProps/custom.xml\": \"application/vnd.openxmlformats-officedocument.custom-properties+xml\",\n      \"/word/_rels/document.xml.rels\": \"application/vnd.openxmlformats-package.relationships+xml\",\n      \"/word/comments.xml\": \"application/vnd.openxmlformats-officedocument.wordprocessingml.comments+xml\",\n      \"/word/commentsExtended.xml\": \"application/vnd.openxmlformats-officedocument.wordprocessingml.commentsExtended+xml\",\n      \"/word/document.xml\": \"application/vnd.openxmlformats-officedocument.wordprocessingml.document.main+xml\",\n      \"/word/fontTable.xml\": \"application/vnd.openxmlformats-officedocument.wordprocessingml.fontTable+xml\",\n      \"/word/numbering.xml\": \"application/vnd.openxmlformats-officedocument.wordprocessingml.numbering+xml\",\n      \"/word/settings.xml\": \"application/vnd.openxmlformats-officedocument.wordprocessingml.settings+xml\",\n      \"/word/styles.xml\": \"application/vnd.openxmlformats-officedocument.wordprocessingml.styles+xml\"\n    },\n    \"web_extension_count\": 1,\n    \"custom_xml_count\": 1,\n    \"header_count\": 0,\n    \"footer_count\": 0\n  },\n  \"rels\": {\n    \"rels\": [\n      [\n        \"http://schemas.openxmlformats.org/package/2006/relationships/metadata/core-properties\",\n        \"rId1\",\n        \"docProps/core.xml\"\n      ],\n      [\n        \"http://schemas.openxmlformats.org/officeDocument/2006/relationships/extended-properties\",\n        \"rId2\",\n        \"docProps/app.xml\"\n      ],\n      [\n        \"http://schemas.openxmlformats.org/officeDocument/2006/relationships/officeDocument\",\n        \"rId3\",\n        \"word/document.xml\"\n      ],\n      [\n        \"http://schemas.openxmlformats.org/officeDocument/2006/relationships/custom-properties\",\n        \"rId4\",\n        \"docProps/custom.xml\"\n      ]\n    ]\n  },\n  \"documentRels\": {\n    \"hasComments\": false,\n    \"hasNumberings\": false,\n    \"images\": [],\n    \"hyperlinks\": [],\n    \"customXmlCount\": 0,\n    \"headerCount\": 0,\n    \"footerCount\": 0\n  },\n  \"docProps\": {\n    \"app\": {},\n    \"core\": {\n      \"config\": {\n        \"created\": null,\n        \"creator\": null,\n        \"description\": null,\n        \"language\": null,\n        \"lastModifiedBy\": null,\n        \"modified\": null,\n        \"revision\": null,\n        \"subject\": null,\n        \"title\": null\n      }\n    },\n    \"custom\": {\n      \"properties\": {}\n    }\n  },\n  \"styles\": {\n    \"docDefaults\": {\n      \"runPropertyDefault\": {\n        \"runProperty\": {\n          \"sz\": 21,\n          \"szCs\": 21,\n          \"fonts\": {\n            \"asciiTheme\": \"minorHAnsi\",\n            \"hiAnsiTheme\": \"minorHAnsi\",\n            \"eastAsiaTheme\": \"minorEastAsia\",\n            \"csTheme\": \"minorBidi\"\n          }\n        }\n      }\n    },\n    \"styles\": [\n      {\n        \"styleId\": \"Normal\",\n        \"name\": \"Normal\",\n        \"styleType\": \"paragraph\",\n        \"runProperty\": {},\n        \"paragraphProperty\": {\n          \"runProperty\": {},\n          \"alignment\": \"both\",\n          \"tabs\": []\n        },\n        \"tableProperty\": {\n          \"width\": {\n            \"width\": 0,\n            \"widthType\": \"auto\"\n          },\n          \"justification\": \"left\",\n          \"borders\": {\n            \"top\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"top\",\n              \"space\": 0\n            },\n            \"left\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"left\",\n              \"space\": 0\n            },\n            \"bottom\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"bottom\",\n              \"space\": 0\n            },\n            \"right\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"right\",\n              \"space\": 0\n            },\n            \"insideH\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"insideH\",\n              \"space\": 0\n            },\n            \"insideV\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"insideV\",\n              \"space\": 0\n            }\n          }\n        },\n        \"tableCellProperty\": {\n          \"width\": null,\n          \"borders\": null,\n          \"gridSpan\": null,\n          \"verticalMerge\": null,\n          \"verticalAlign\": null,\n          \"textDirection\": null,\n          \"shading\": null\n        },\n        \"basedOn\": null,\n        \"next\": null\n      },\n      {\n        \"styleId\": \"DefaultParagraphFont\",\n        \"name\": \"Default Paragraph Font\",\n        \"styleType\": \"character\",\n        \"runProperty\": {},\n        \"paragraphProperty\": {\n          \"runProperty\": {},\n          \"tabs\": []\n        },\n        \"tableProperty\": {\n          \"width\": {\n            \"width\": 0,\n            \"widthType\": \"auto\"\n          },\n          \"justification\": \"left\",\n          \"borders\": {\n            \"top\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"top\",\n              \"space\": 0\n            },\n            \"left\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"left\",\n              \"space\": 0\n            },\n            \"bottom\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"bottom\",\n              \"space\": 0\n            },\n            \"right\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"right\",\n              \"space\": 0\n            },\n            \"insideH\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"insideH\",\n              \"space\": 0\n            },\n            \"insideV\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"insideV\",\n              \"space\": 0\n            }\n          }\n        },\n        \"tableCellProperty\": {\n          \"width\": null,\n          \"borders\": null,\n          \"gridSpan\": null,\n          \"verticalMerge\": null,\n          \"verticalAlign\": null,\n          \"textDirection\": null,\n          \"shading\": null\n        },\n        \"basedOn\": null,\n        \"next\": null\n      },\n      {\n        \"styleId\": \"TableNormal\",\n        \"name\": \"Normal Table\",\n        \"styleType\": \"table\",\n        \"runProperty\": {},\n        \"paragraphProperty\": {\n          \"runProperty\": {},\n          \"tabs\": []\n        },\n        \"tableProperty\": {\n          \"width\": {\n            \"width\": 0,\n            \"widthType\": \"auto\"\n          },\n          \"justification\": \"left\",\n          \"borders\": {\n            \"top\": null,\n            \"left\": null,\n            \"bottom\": null,\n            \"right\": null,\n            \"insideH\": null,\n            \"insideV\": null\n          },\n          \"margins\": {\n            \"top\": {\n              \"val\": 0,\n              \"widthType\": \"dxa\"\n            },\n            \"left\": {\n              \"val\": 108,\n              \"widthType\": \"dxa\"\n            },\n            \"bottom\": {\n              \"val\": 0,\n              \"widthType\": \"dxa\"\n            },\n            \"right\": {\n              \"val\": 108,\n              \"widthType\": \"dxa\"\n            }\n          }\n        },\n        \"tableCellProperty\": {\n          \"width\": null,\n          \"borders\": null,\n          \"gridSpan\": null,\n          \"verticalMerge\": null,\n          \"verticalAlign\": null,\n          \"textDirection\": null,\n          \"shading\": null\n        },\n        \"basedOn\": null,\n        \"next\": null\n      },\n      {\n        \"styleId\": \"NoList\",\n        \"name\": \"No List\",\n        \"styleType\": \"numbering\",\n        \"runProperty\": {},\n        \"paragraphProperty\": {\n          \"runProperty\": {},\n          \"tabs\": []\n        },\n        \"tableProperty\": {\n          \"width\": {\n            \"width\": 0,\n            \"widthType\": \"auto\"\n          },\n          \"justification\": \"left\",\n          \"borders\": {\n            \"top\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"top\",\n              \"space\": 0\n            },\n            \"left\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"left\",\n              \"space\": 0\n            },\n            \"bottom\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"bottom\",\n              \"space\": 0\n            },\n            \"right\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"right\",\n              \"space\": 0\n            },\n            \"insideH\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"insideH\",\n              \"space\": 0\n            },\n            \"insideV\": {\n              \"borderType\": \"single\",\n              \"size\": 2,\n              \"color\": \"000000\",\n              \"position\": \"insideV\",\n              \"space\": 0\n            }\n          }\n        },\n        \"tableCellProperty\": {\n          \"width\": null,\n          \"borders\": null,\n          \"gridSpan\": null,\n          \"verticalMerge\": null,\n          \"verticalAlign\": null,\n          \"textDirection\": null,\n          \"shading\": null\n        },\n        \"basedOn\": null,\n        \"next\": null\n      }\n    ]\n  },\n  \"document\": {\n    \"children\": [\n      {\n        \"type\": \"paragraph\",\n        \"data\": {\n          \"id\": \"5BADD4ED\",\n          \"children\": [\n            {\n              \"type\": \"bookmarkStart\",\n              \"data\": {\n                \"id\": 0,\n                \"name\": \"_GoBack\"\n              }\n            },\n            {\n              \"type\": \"bookmarkEnd\",\n              \"data\": {\n                \"id\": 0\n              }\n            },\n            {\n              \"type\": \"run\",\n              \"data\": {\n                \"runProperty\": {},\n                \"children\": [\n                  {\n                    \"type\": \"text\",\n                    \"data\": {\n                      \"preserveSpace\": true,\n                      \"text\": \"Hello\"\n                    }\n                  }\n                ]\n              }\n            }\n          ],\n          \"property\": {\n            \"runProperty\": {},\n            \"tabs\": []\n          },\n          \"hasNumbering\": false\n        }\n      }\n    ],\n    \"sectionProperty\": {\n      \"pageSize\": {\n        \"w\": 11906,\n        \"h\": 16838,\n        \"orient\": null\n      },\n      \"pageMargin\": {\n        \"top\": 1440,\n        \"left\": 1440,\n        \"bottom\": 1440,\n        \"right\": 1440,\n        \"header\": 720,\n        \"footer\": 720,\n        \"gutter\": 0\n      },\n      \"columns\": 1,\n      \"space\": 425,\n      \"titlePg\": false,\n      \"textDirection\": \"lrTb\",\n      \"docGrid\": {\n        \"gridType\": \"lines\",\n        \"linePitch\": 360,\n        \"charSpace\": null\n      }\n    },\n    \"hasNumbering\": false\n  },\n  \"comments\": {\n    \"comments\": []\n  },\n  \"numberings\": {\n    \"abstractNums\": [],\n    \"numberings\": []\n  },\n  \"settings\": {\n    \"defaultTabStop\": 840,\n    \"zoom\": 100,\n    \"docId\": \"58f67304-63b1-4505-b4ba-0c1a55a3ad31\",\n    \"docVars\": [],\n    \"evenAndOddHeaders\": false,\n    \"adjustLineHeightInTable\": true\n  },\n  \"fontTable\": {},\n  \"media\": [],\n  \"commentsExtended\": {\n    \"children\": []\n  },\n  \"webSettings\": {\n    \"divs\": []\n  },\n  \"taskpanes\": null,\n  \"taskpanesRels\": {\n    \"rels\": []\n  },\n  \"webExtensions\": [],\n  \"customItems\": [],\n  \"customItemProps\": [],\n  \"customItemRels\": [],\n  \"themes\": [\n    {\n      \"fontSchema\": {\n        \"majorFont\": {\n          \"latin\": \"Arial\",\n          \"ea\": \"\",\n          \"cs\": \"\",\n          \"fonts\": [\n            {\n              \"script\": \"Jpan\",\n              \"typeface\": \"ＭＳ ゴシック\"\n            },\n            {\n              \"script\": \"Hang\",\n              \"typeface\": \"맑은 고딕\"\n            },\n            {\n              \"script\": \"Hans\",\n              \"typeface\": \"宋体\"\n            },\n            {\n              \"script\": \"Hant\",\n              \"typeface\": \"新細明體\"\n            },\n            {\n              \"script\": \"Arab\",\n              \"typeface\": \"Times New Roman\"\n            },\n            {\n              \"script\": \"Hebr\",\n              \"typeface\": \"Times New Roman\"\n            },\n            {\n              \"script\": \"Thai\",\n              \"typeface\": \"Angsana New\"\n            },\n            {\n              \"script\": \"Ethi\",\n              \"typeface\": \"Nyala\"\n            },\n            {\n              \"script\": \"Beng\",\n              \"typeface\": \"Vrinda\"\n            },\n            {\n              \"script\": \"Gujr\",\n              \"typeface\": \"Shruti\"\n            },\n            {\n              \"script\": \"Khmr\",\n              \"typeface\": \"MoolBoran\"\n            },\n            {\n              \"script\": \"Knda\",\n              \"typeface\": \"Tunga\"\n            },\n            {\n              \"script\": \"Guru\",\n              \"typeface\": \"Raavi\"\n            },\n            {\n              \"script\": \"Cans\",\n              \"typeface\": \"Euphemia\"\n            },\n            {\n              \"script\": \"Cher\",\n              \"typeface\": \"Plantagenet Cherokee\"\n            },\n            {\n              \"script\": \"Yiii\",\n              \"typeface\": \"Microsoft Yi Baiti\"\n            },\n            {\n              \"script\": \"Tibt\",\n              \"typeface\": \"Microsoft Himalaya\"\n            },\n            {\n              \"script\": \"Thaa\",\n              \"typeface\": \"MV Boli\"\n            },\n            {\n              \"script\": \"Deva\",\n              \"typeface\": \"Mangal\"\n            },\n            {\n              \"script\": \"Telu\",\n              \"typeface\": \"Gautami\"\n            },\n            {\n              \"script\": \"Taml\",\n              \"typeface\": \"Latha\"\n            },\n            {\n              \"script\": \"Syrc\",\n              \"typeface\": \"Estrangelo Edessa\"\n            },\n            {\n              \"script\": \"Orya\",\n              \"typeface\": \"Kalinga\"\n            },\n            {\n              \"script\": \"Mlym\",\n              \"typeface\": \"Kartika\"\n            },\n            {\n              \"script\": \"Laoo\",\n              \"typeface\": \"DokChampa\"\n            },\n            {\n              \"script\": \"Sinh\",\n              \"typeface\": \"Iskoola Pota\"\n            },\n            {\n              \"script\": \"Mong\",\n              \"typeface\": \"Mongolian Baiti\"\n            },\n            {\n              \"script\": \"Viet\",\n              \"typeface\": \"Times New Roman\"\n            },\n            {\n              \"script\": \"Uigh\",\n              \"typeface\": \"Microsoft Uighur\"\n            },\n            {\n              \"script\": \"Geor\",\n              \"typeface\": \"Sylfaen\"\n            }\n          ]\n        },\n        \"minorFont\": {\n          \"latin\": \"Century\",\n          \"ea\": \"\",\n          \"cs\": \"\",\n          \"fonts\": [\n            {\n              \"script\": \"Jpan\",\n              \"typeface\": \"ＭＳ 明朝\"\n            },\n            {\n              \"script\": \"Hang\",\n              \"typeface\": \"맑은 고딕\"\n            },\n            {\n              \"script\": \"Hans\",\n              \"typeface\": \"宋体\"\n            },\n            {\n              \"script\": \"Hant\",\n              \"typeface\": \"新細明體\"\n            },\n            {\n              \"script\": \"Arab\",\n              \"typeface\": \"Arial\"\n            },\n            {\n              \"script\": \"Hebr\",\n              \"typeface\": \"Arial\"\n            },\n            {\n              \"script\": \"Thai\",\n              \"typeface\": \"Cordia New\"\n            },\n            {\n              \"script\": \"Ethi\",\n              \"typeface\": \"Nyala\"\n            },\n            {\n              \"script\": \"Beng\",\n              \"typeface\": \"Vrinda\"\n            },\n            {\n              \"script\": \"Gujr\",\n              \"typeface\": \"Shruti\"\n            },\n            {\n              \"script\": \"Khmr\",\n              \"typeface\": \"DaunPenh\"\n            },\n            {\n              \"script\": \"Knda\",\n              \"typeface\": \"Tunga\"\n            },\n            {\n              \"script\": \"Guru\",\n              \"typeface\": \"Raavi\"\n            },\n            {\n              \"script\": \"Cans\",\n              \"typeface\": \"Euphemia\"\n            },\n            {\n              \"script\": \"Cher\",\n              \"typeface\": \"Plantagenet Cherokee\"\n            },\n            {\n              \"script\": \"Yiii\",\n              \"typeface\": \"Microsoft Yi Baiti\"\n            },\n            {\n              \"script\": \"Tibt\",\n              \"typeface\": \"Microsoft Himalaya\"\n            },\n            {\n              \"script\": \"Thaa\",\n              \"typeface\": \"MV Boli\"\n            },\n            {\n              \"script\": \"Deva\",\n              \"typeface\": \"Mangal\"\n            },\n            {\n              \"script\": \"Telu\",\n              \"typeface\": \"Gautami\"\n            },\n            {\n              \"script\": \"Taml\",\n              \"typeface\": \"Latha\"\n            },\n            {\n              \"script\": \"Syrc\",\n              \"typeface\": \"Estrangelo Edessa\"\n            },\n            {\n              \"script\": \"Orya\",\n              \"typeface\": \"Kalinga\"\n            },\n            {\n              \"script\": \"Mlym\",\n              \"typeface\": \"Kartika\"\n            },\n            {\n              \"script\": \"Laoo\",\n              \"typeface\": \"DokChampa\"\n            },\n            {\n              \"script\": \"Sinh\",\n              \"typeface\": \"Iskoola Pota\"\n            },\n            {\n              \"script\": \"Mong\",\n              \"typeface\": \"Mongolian Baiti\"\n            },\n            {\n              \"script\": \"Viet\",\n              \"typeface\": \"Arial\"\n            },\n            {\n              \"script\": \"Uigh\",\n              \"typeface\": \"Microsoft Uighur\"\n            },\n            {\n              \"script\": \"Geor\",\n              \"typeface\": \"Sylfaen\"\n            }\n          ]\n        }\n      }\n    }\n  ],\n  \"images\": [],\n  \"hyperlinks\": []\n}"
